### PR TITLE
catalog-ms-graph: fixes #13298, ignored userExpand and groupExpand parameters

### LINKED ISSUE #13298: ms-graph plugin: expand not working on MicrosoftGraphOrgEntityProvider
## Expected Behavior

Documented configuration features userExpand and groupExpand should expand the requested properties

```yaml
user:
  # Optional parameter to include the expanded resource or collection referenced
  # by a single relationship (navigation property) in your results.
  # Only one relationship can be expanded in a single request.
  # See https://docs.microsoft.com/en-us/graph/query-parameters#expand-parameter
  # Can be combined with userGroupMember[...] instead of userFilter.
  expand: manager
```

## Actual Behavior

Does not work. New MicrosoftGraphOrgEntityProvider doesn't pass configuration in main ingestion loop:
```javascript
async read(options?: { logger?: Logger }) {
    if (!this.connection) {
      throw new Error('Not initialized');
    }

    const logger = options?.logger ?? this.options.logger;
    const provider = this.options.provider;
    const { markReadComplete } = trackProgress(logger);
    const client = MicrosoftGraphClient.create(this.options.provider);
    const { users, groups } = await readMicrosoftGraphOrg(
      client,
      provider.tenantId,
      {
        userFilter: provider.userFilter,
        userGroupMemberFilter: provider.userGroupMemberFilter,
        userGroupMemberSearch: provider.userGroupMemberSearch,
        groupFilter: provider.groupFilter,
        groupSearch: provider.groupSearch,
        groupSelect: provider.groupSelect,
        queryMode: provider.queryMode,
        groupTransformer: this.options.groupTransformer,
        userTransformer: this.options.userTransformer,
        organizationTransformer: this.options.organizationTransformer,
        logger: logger,
      },
    );
```

## Steps to Reproduce

1. Set userExpand configuration to "memberOf"
2. Use new ingestor MicrosoftGraphOrgEntityProvider
3. New ingestor does not expand property (view in debugger or in transformer)

## Context

I am trying to add custom information to user entities via the userExpand configuration parameter

## Your Environment

Latest Backstage, Latest plugin version

### PR DESCRIPTION
**Pull request by gtestault**
## Hey, I just made a Pull Request!

fixes #13298

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: .changeset/afraid-radios-boil.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-catalog-backend-module-msgraph': patch
+---
+
+Fixed bug in MicrosoftGraphEntityProvider ignores userExpand and groupExpand configuration parameters
```
Comment (jhaals) [inline .changeset/afraid-radios-boil.md:5]:
```suggestion
Fixed a bug in the `MicrosoftGraphEntityProvider` that ignored the `userExpand` and `groupExpand` configuration parameters
```

**Review by jhaals — CHANGES_REQUESTED** [review]
Thanks for the fix! Needs a slight formatting change for the changeset.

**Review by jhaals — APPROVED** [review]

**Comment by jhaals** [comment]
Thanks!

**Comment by suuus** [comment]
Congrats on your first PR merged @gtestault! Happy to count you as a contributor.

**Pull request merged**